cclient/iota_client_extended_api.c

```diff
@@ -249,3 +249,80 @@ retcode_t iota_client_get_account_data(iota_client_service_t const* const serv,
   get_balances_res_free(&balances_res);
   return ret_code;
 }
+
+retcode_t iota_client_get_bundle(iota_client_service_t const* const serv,
+                                 flex_trit_t const* const tail_hash,
+                                 transaction_objs_t* out_tx_objs) {
+  retcode_t ret_code = RC_OK;
+
+  log_debug(CCLIENT_EXTENDED_LOGGER_ID, "[%s:%d]\n", __func__, __LINE__);
+
+  return iota_client_traverse_bundle(serv, tail_hash, out_tx_objs);
+}
+
+retcode_t iota_client_traverse_bundle(iota_client_service_t const* const serv,
+                                      flex_trit_t const* const trunk_tx,
+                                      transaction_objs_t *const tx_objs) {
+  retcode_t ret_code = RC_OK;
+  get_trytes_req_t* get_trytes_req = NULL;
+  get_trytes_res_t* get_trytes_res = NULL;
+  iota_transaction_t tx;
+  flex_trit_t* tmp_trytes;
+  flex_trit_t* trunk_hash = (flex_trit_t *)trunk_tx;
+  
+  log_debug(CCLIENT_EXTENDED_LOGGER_ID, "[%s:%d]\n", __func__, __LINE__);
+
+  do {
+    // Create request and response to get_trytes API
+    get_trytes_req = get_trytes_req_new();
+    get_trytes_res = get_trytes_res_new();
+    if (!get_trytes_req || !get_trytes_res) {
+      ret_code = RC_CCLIENT_NULL_PTR;
+      goto cleanup;
+    }
+    // Add the trunk hash to the request parameters
+    ret_code = hash243_queue_push(&get_trytes_req->hashes, trunk_hash);
+    if (ret_code != RC_OK) {
+      goto cleanup;
+    }
+    // Call the get_trytes API
+    ret_code = iota_client_get_trytes(serv, get_trytes_req, get_trytes_res);
+    if (ret_code != RC_OK) {
+      goto cleanup;
+    }
+    get_trytes_req_free(&get_trytes_req);
+    // Get the transaction trytes
+    tmp_trytes = hash8019_queue_at(&get_trytes_res->trytes, 0);
+    if (!tmp_trytes) {
+      ret_code = RC_CCLIENT_RES_ERROR;
+      goto cleanup;
+    }
+    // Create a new transaction with the received trytes
+    tx = transaction_deserialize(tmp_trytes);
```

Comment:
`transaction_deserialize` does a dynamic allocation. No need for that here ? Use a statically allocated transaction object + `transaction_deserialize_from_trits`

Reply:
We potentially want to return several transactions

Reply:
But `tx` will be copied into the array right ? So no need to have it dynamic ?